### PR TITLE
fix OSX 32 build

### PR DESCRIPTION
**Pull request by WalterBright**
Because it broke again.

### CODE REVIEW

**Comment by yebblies** [comment]
Are you the only person using OSX32?

File: src/backend/type.h [hunk 1]
```diff
@@ -206,7 +206,7 @@ type *type_dyn_array(type *tnext);
 extern "C" type *type_static_array(targ_size_t dim, type *tnext);
 type *type_assoc_array(type *tkey, type *tvalue);
 type *type_delegate(type *tnext);
-type *type_function(tym_t tyf, type **ptypes, size_t nparams, bool variadic, type *tret);
+extern "C" { type *type_function(tym_t tyf, type **ptypes, size_t nparams, bool variadic, type *tret); }
```
Comment (yebblies) [inline src/backend/type.h]: Don't need braces.

**Comment by WalterBright** [comment]
Quite possibly. But as I pointed out last time this came up, these changes make it work for any future port that may have a peculiar typedef for size_t.

**Comment by yebblies** [comment]
That would be a more important concern if we were planning to keep the backend in C++ long term.  At the moment we don't have any officially supported backends that need this and we are unlikely to ever add any.

Anyway, I'll pull this with the braces change.

**Comment by WalterBright** [comment]
> I'll pull this with the braces change.

done.

**Pull request merged**